wcfsetup/install/files/lib/system/language/preload/LanguagePreloader.class.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace wcf\system\language\preload;
+
+use wcf\data\language\Language;
+use wcf\system\event\EventHandler;
+use wcf\system\Regex;
+use wcf\util\DirectoryUtil;
+use wcf\util\FileUtil;
+
+final class LanguagePreloader
+{
+    private readonly EventHandler $eventHandler;
+    private readonly Language $language;
+
+    private const STORAGE_PATH = \WCF_DIR . 'js/preload/';
+
+    public function __construct(Language $language)
+    {
+        $this->eventHandler = EventHandler::getInstance();
+        $this->language = $language;
```

Comment:
`LanguagePreloader` is effectively a service and thus should avoid state. This is important for a possible future migration to dependency injection.

The `Language` parameter should be moved into the `rebuild()` function and then passed along.

Alternatively this could be transformed into a command pattern where the dependency injection issue is solved by the introduction of a command bus. In that case there should only be a single public `__invoke()` method.

Reply:
> LanguagePreloader is effectively a service and thus should avoid state. This is important for a possible future migration to dependency injection.

More specifically: Yes, it should avoid state as much as possible. However it absolutely must not be parameterized, treat it as if it was a singleton.